### PR TITLE
add new UpdateCaseDataJob job

### PR DESCRIPTION
**Pull request by iguannago**
https://tools.hmcts.net/jira/browse/RWA-662

### CODE REVIEW

**Review by indvamshi — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import uk.gov.hmcts.reform.wataskmonitor.domain.idam.UserInfo;
+import uk.gov.hmcts.reform.wataskmonitor.services.UpdateCaseDataService;
+
+@Component
+public class ManagementCategoryDataService implements UpdateCaseDataService {
+
+    public static final String EVEN_DESCRIPTION =
+        "caseManagementCategory data added by WA-TASK-MONITOR service and AD_HOC_UPDATE_CASE_DATA job";
+    public static final String EVENT_SUMMARY = "caseManagementCategory data added by WA-TASK-MONITOR service";
+    public static final String EVENT_ID = "adminCaseUpdate";
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java:19]: 👍

**Review by raghera — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.wataskmonitor.config.idam.IdamTokenGenerator;
+import uk.gov.hmcts.reform.wataskmonitor.domain.idam.UserInfo;
+import uk.gov.hmcts.reform.wataskmonitor.services.UpdateCaseDataService;
+
+@Component
+public class ManagementCategoryDataService implements UpdateCaseDataService {
```
[the remark this answers is not in the document]
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java:14]: done

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    public static final String EVEN_DESCRIPTION =
+        "caseManagementCategory data added by WA-TASK-MONITOR service and AD_HOC_UPDATE_CASE_DATA job";
+    public static final String EVENT_SUMMARY = "caseManagementCategory data added by WA-TASK-MONITOR service";
+    public static final String EVENT_ID = "editAppealAfterSubmit";
```
[the remark this answers is not in the document]
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java]: that's old. It is the `adminCaseUpdate`

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    @Override
+    public boolean updateCaseInCcd(String caseId, String serviceToken) {
+        String userToken = systemUserIdamToken.generate();
+        UserInfo userInfo = systemUserIdamToken.getUserInfo(userToken);
+
+        StartEventResponse startEventResponse = coreCaseDataApi.startEventForCaseWorker(
+            userToken,
+            serviceToken,
+            userInfo.getUid(),
+            "IA",
```
[the remark this answers is not in the document]
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java:40]: Yes, this is the implementation for this particular job. If tomorrow we need to make it for SSCS then we can either add a new implementation or even better, if we could update the interface to allow a generic param to make it more flexible, yes.

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            CaseDataContent.builder()
+                .event(Event.builder()
+                           .id(EVENT_ID)
+                           .description(EVEN_DESCRIPTION)
+                           .summary(EVENT_SUMMARY)
+                           .build())
+                .eventToken(startEventResponse.getToken())
+                .build()
+        );
+        return result.getData().get(CASE_MANAGEMENT_CATEGORY_FIELD_NAME) != null;
```
[the remark this answers is not in the document]
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/updatecasedata/ManagementCategoryDataService.java:63]: Yes, I put the changes to meet the ticket ACs. If we need that tomorrow, we can update it easily

**Review by indvamshi — APPROVED** [review]

**Pull request merged**